core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -136,8 +141,43 @@ private static XContentBuilder build(RestChannel channel, RestStatus status, Exc
 
         XContentBuilder builder = channel.newErrorBuilder().startObject();
         ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
-        builder.field("status", status.getStatus());
+        builder.field(STATUS, status.getStatus());
         builder.endObject();
         return builder;
     }
+
+    public static ElasticsearchStatusException errorFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        ElasticsearchException exception = null;
+        RestStatus status = null;
+
+        String currentFieldName = parser.currentName();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            }
+            if (STATUS.equals(currentFieldName)) {
+                if (token.isValue()) {
+                    status = RestStatus.fromCode(parser.intValue());
+                }
```

Comment:
It shouldn't be something else than a number value, so I changed this to:
```
if (STATUS.equals(currentFieldName)) {
    if (token != XContentParser.Token.FIELD_NAME) {
        ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token,  ...);
        status = RestStatus.fromCode(parser.intValue());
    }
 } else {
     exception = ElasticsearchException.failureFromXContent(parser);
}
```

which is I think more coherent with what we do in other parsing methods.